pkg/glusterfs/controllerserver.go

```diff
@@ -53,18 +54,78 @@ type ProvisionerConfig struct {
 	csiConf  *CsiDrvParam
 }
 
+func convertVolumeParam(volumeString string) (int, error) {
+
+	count, err := strconv.Atoi(volumeString)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse volumestring %q: %v", volumeString, err)
+	}
+
+	if count < 0 {
+		return 0, fmt.Errorf("negative values are not allowed")
+	}
+	return count, nil
+}
+
 // ParseCreateVolRequest parse incoming volume create request and fill ProvisionerConfig.
 func (cs *ControllerServer) ParseCreateVolRequest(req *csi.CreateVolumeRequest) (*ProvisionerConfig, error) {
 
 	var reqConf ProvisionerConfig
 	var gdReq api.VolCreateReq
-
+	parseVolumeType := ""
 	reqConf.gdVolReq = &gdReq
 
 	// Get Volume name
 	if req != nil {
 		reqConf.gdVolReq.Name = req.Name
 	}
+
+	for k, v := range req.GetParameters() {
```

Comment:
what is the value we can expect in `GetParameters()` 
as this is a map I think we should not return the error if we hit default case while parsing the map.
if the user adds some more extra keys to the storage class I think this will fail.

Reply:
@Madhu-1 thats correct, if user put extra keywords which are not known to driver, it should fail for sure.